smalld/smalld.py

```diff
@@ -205,9 +207,11 @@ def send_request(self, method, path, payload="", attachments=None):
         else:
             args = {}
 
-        r = self.session.request(method, f"{self.base_url}/{path}", **args)
+        self.limiter.on_request(method, path)
+        res = self.session.request(method, f"{self.base_url}/{path}", **args)
```

Comment:
What does requests to on non-200 responses? Does it throw an exception or just give you the response?

Reply:
It returns the response, for raising an exception you just call `res.raise_for_status()`.